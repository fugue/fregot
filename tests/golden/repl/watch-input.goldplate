{
    "command": "bash",
    "arguments": ["${GOLDPLATE_NAME}.sh"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"stdout": "${GOLDPLATE_NAME}.stdout"}
    ]
}
