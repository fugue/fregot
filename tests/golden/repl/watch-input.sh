#!/bin/bash
set -o nounset -o errexit -o pipefail

input1='watch-input-1.json'
input2='watch-input-2.json'
pipe="$(mktemp -u)"

trap "rm -f \"$pipe\" \"$input1\" \"$input2\"" EXIT

echo '{"name": "input-1"}' >"$input1"
echo '{"name": "input-2"}' >"$input2"
mkfifo "$pipe"

sleep 500 >"$pipe" 2>/dev/null &  # Just to keep this file open.
doorstop_pid=$!

fregot repl --no-history-file --watch <"$pipe" &
fregot_pid="$!"

function enter() {
    echo "$1" >>"$pipe"
    sleep 0.1
}

enter ":input $input1"
enter "input"
enter ":input $input2"
enter "input"
echo '{"name": "input-changed!"}' >"$input2"
sleep 0.1
enter "input"

kill -PIPE "$doorstop_pid"
