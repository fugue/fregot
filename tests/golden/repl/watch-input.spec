{
    "command": "bash",
    "arguments": ["${SPEC_NAME}.sh"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"stdout": "${SPEC_NAME}.stdout"}
    ]
}
