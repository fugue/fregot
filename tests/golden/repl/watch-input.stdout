repl% repl% = [33m{[0m[31m"name"[0m[33m:[0m [31m"input-1"[0m[33m}[0m
repl% repl% = [33m{[0m[31m"name"[0m[33m:[0m [31m"input-2"[0m[33m}[0m
repl% 
repl% = [33m{[0m[31m"name"[0m[33m:[0m [31m"input-changed!"[0m[33m}[0m
repl% 